platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/SupportMapFragment.java

```diff
@@ -76,9 +82,18 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
   public void onViewCreated(View view, Bundle savedInstanceState) {
     super.onViewCreated(view, savedInstanceState);
     map.onCreate(savedInstanceState);
-    if (onMapReadyCallback != null) {
-      map.getMapAsync(onMapReadyCallback);
+    map.getMapAsync(this);
+  }
+
+  @Override
+  public void onMapReady(MapboxMap mapboxMap) {
+    this.mapboxMap = mapboxMap;
+    if (!mapReadyCallbackList.isEmpty()) {
```

Comment:
This check's redundant. Besides that, the same question as above about the list.

Reply:
This check is not redundant, checking if a list is not empty before looping it will avoid creating an iterator. In code path that are called often this can make a difference. In this case though it doesn't result in any significant memory growth since it's only called once. I will remove it.